modules/solid_mechanics/src/materials/ADComputeShellStress.C

```diff
@@ -54,11 +58,24 @@ ADComputeShellStress::ADComputeShellStress(const InputParameters & parameters)
     _strain_increment[t] =
         &getADMaterialProperty<RankTwoTensor>("strain_increment_t_points_" + std::to_string(t));
     // rotation matrix and stress for output purposes only
+    _local_transformation_matrix[t] =
+        &getMaterialProperty<RankTwoTensor>("local_transformation_t_points_" + std::to_string(t));
     _covariant_transformation_matrix[t] = &getMaterialProperty<RankTwoTensor>(
         "covariant_transformation_t_points_" + std::to_string(t));
     _global_stress[t] =
         &declareProperty<RankTwoTensor>("global_stress_t_points_" + std::to_string(t));
+    _local_stress[t] =
+        &declareProperty<RankTwoTensor>("local_stress_t_points_" + std::to_string(t));
   }
+
+  _shell_force_1 = &declareProperty<Real>("shell_force_1");
```

Comment:
I'm not sure I like the idea of declaring and computing these properties all the time in this class since they're just used for output. What would you think about making an elemental AuxKernel that computes the average resultants? Since you have to add an AuxKernel to output the averages of these material properties, it wouldn't be any more work than you're already doing in the input file.

Reply:
I added an AuxKernel (ShellResultantsAux) that stores the resultants in Aux variables.